packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -2,42 +2,164 @@
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	privateApis as componentsPrivateApis,
 } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
-import { canBindAttribute } from '../hooks/use-bindings-attributes';
+import {
+	canBindAttribute,
+	getBindableAttributes,
+} from '../hooks/use-bindings-attributes';
+import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
 import InspectorControls from '../components/inspector-controls';
+import BlockContext from '../components/block-context';
+
+const {
+	DropdownMenuV2: DropdownMenu,
+	DropdownMenuGroupV2: DropdownMenuGroup,
+	DropdownMenuCheckboxItemV2: DropdownMenuCheckboxItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+	DropdownMenuItemHelpTextV2: DropdownMenuItemHelpText,
+	DropdownMenuSeparatorV2: DropdownMenuSeparator,
+} = unlock( componentsPrivateApis );
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
+
+function BlockBindingsPanelDropdown( {
+	fieldsList,
+	addConnection,
+	removeConnection,
+	attribute,
+	binding,
+} ) {
+	const currentKey = binding?.args?.key;
+	return (
+		<>
+			{ Object.entries( fieldsList ).map( ( [ label, fields ] ) => (
+				<>
+					<DropdownMenuGroup key={ label }>
+						{ Object.keys( fieldsList ).length > 1 && (
+							<Text
+								className="block-editor-bindings__source-label"
+								upperCase
+								variant="muted"
+							>
+								{ label }
+							</Text>
+						) }
+						{ Object.entries( fields ).map( ( [ key, value ] ) => (
+							<DropdownMenuCheckboxItem
+								key={ key }
+								onClick={ () =>
+									addConnection( key, attribute )
+								}
+								name={ attribute + '-binding' }
+								value={ key }
+								checked={ key === currentKey }
+							>
+								<DropdownMenuItemLabel numberOfLines={ 1 }>
+									{ key }
+								</DropdownMenuItemLabel>
+								<DropdownMenuItemHelpText numberOfLines={ 1 }>
+									{ value }
+								</DropdownMenuItemHelpText>
+							</DropdownMenuCheckboxItem>
+						) ) }
+					</DropdownMenuGroup>
+					<DropdownMenuSeparator />
+				</>
+			) ) }
+			<ButtonGroup className="block-editor-bindings__clear-binding">
+				<Button
+					__next40pxDefaultSize
+					disabled={ ! binding }
+					accessibleWhenDisabled
+					onClick={ () => removeConnection( attribute ) }
+					variant="tertiary"
+				>
+					{ __( 'Clear' ) }
+				</Button>
+			</ButtonGroup>
```

Comment:
I can remove the clear button for now. I was just experimenting how it could look like.

Reply:
By the way, the "Clear" button is added in other places where the `ToolsPanel` is used. Some examples could be the paragraph colors or the image duotone.

Reply:
I removed it. We can consider adding it properly in a follow-up pull request.

Reply:
> By the way, the "Clear" button is added in other places where the ToolsPanel is used. Some examples could be the paragraph colors or the image duotone.

I wasn't aware of that — I always assumed that `ToolsPanel` would take care of resetting controls via the ellipsis menu, instead of having the individual controls provide a "clear" button, but I may be out of date here (cc @aaronrobertshaw ).

In case you wanted to add a "Clear" button, my advice of using a separator + a standard dropdown menu item still stands.